Lazy/Lazy/ILazy.cs

```diff
@@ -0,0 +1,14 @@
+namespace Lazy;
+/// <summary>
+/// Interface for lazy evaluation of a value.
+/// </summary>
+/// <typeparam name="T"></typeparam>
+public interface ILazy<T>
```

Comment:
```suggestion
public interface ILazy<out T>
```
Если есть возможность указать вариантность, надо указать, это упростит пользователям работу с Lazy